app/javascript/controllers/qrscanner_controller.js

```diff
@@ -0,0 +1,51 @@
+import QrScanner from 'qr-scanner';
+
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+
+  initialize() {
+    this.qrScanner = null;
+  }
+
+  _isPositiveInteger(string) {
+    const castedString = Number(string);
+    return !(castedString == NaN || castedString < 0 || !Number.isInteger(castedString))
```

Comment:
Do you want to call `!Number.isInteger(castedString)` before `castedString < 0`? I'm unsure what happens if it isn't a number and you want to test for `< 0`.

Reply:
sounds like a good idea!